payments/models.py

```diff
@@ -0,0 +1,146 @@
+from decimal import Decimal
+
+from django.core.exceptions import ValidationError
+from django.core.validators import MinValueValidator
+from django.db import models
+from django.db.models import Q, UniqueConstraint
+from django.utils.translation import ugettext_lazy as _
+from enumfields import EnumField
+
+from payments.enums import AdditionalProductType, PeriodType, PriceUnits, ServiceType
+from utils.models import TimeStampedModel, UUIDModel
+
+TAX_PERCENTAGES = [Decimal(x) for x in ("24.00", "10.00")]
+
+DEFAULT_TAX_PERCENTAGE = Decimal("24.0")
+
+
+class AbstractBaseProduct(TimeStampedModel, UUIDModel):
+    price_value = models.DecimalField(
+        verbose_name=_("price"),
+        max_digits=10,
+        decimal_places=2,
+        validators=[MinValueValidator(Decimal("0.01"))],
+    )
+    price_unit = EnumField(PriceUnits, default=PriceUnits.AMOUNT)
+    tax_percentage = models.DecimalField(
+        verbose_name=_("tax percentage"),
+        max_digits=5,
+        decimal_places=2,
+        default=DEFAULT_TAX_PERCENTAGE,
+        choices=[(tax, str(tax)) for tax in TAX_PERCENTAGES],
+    )
+
+    class Meta:
+        abstract = True
+
+    def save(self, *args, **kwargs):
+        self.full_clean()
+        super().save(*args, **kwargs)
+
+
+class BerthPriceGroup(UUIDModel):
+    name = models.CharField(
+        verbose_name=_("berth price group name"), max_length=128, null=True, blank=True
+    )
+
+    def __str__(self):
+        return f"{self.name} ({self.id})"
+
+
+class AbstractPlaceProduct(AbstractBaseProduct):
+    price_unit = EnumField(
+        PriceUnits,
+        choices=[(PriceUnits.AMOUNT, PriceUnits.AMOUNT.label)],
+        default=PriceUnits.AMOUNT,
+    )
+    tax_percentage = models.DecimalField(
+        verbose_name=_("tax percentage"),
+        max_digits=5,
+        decimal_places=2,
+        default=DEFAULT_TAX_PERCENTAGE,
+        choices=[(DEFAULT_TAX_PERCENTAGE, str(DEFAULT_TAX_PERCENTAGE))],
+    )
+
+    class Meta:
+        abstract = True
+
+
+class BerthProduct(AbstractPlaceProduct):
+    price_group = models.ForeignKey(
+        BerthPriceGroup,
+        on_delete=models.CASCADE,
+        verbose_name=_("price group"),
+        related_name=_("products"),
+    )
+    harbor = models.ForeignKey(
+        "resources.Harbor",
+        verbose_name=_("harbor"),
+        related_name=_("product"),
+        on_delete=models.CASCADE,
+    )
+
+    class Meta:
+        constraints = [
+            UniqueConstraint(
+                fields=["price_group", "harbor"],
+                name="unique_product_for_harbor_pricegroup",
+            ),
+        ]
+
+    def __str__(self):
+        return f"{self.price_group} - {self.harbor} ({self.price_value}€)"
+
+
+class WinterStorageProduct(AbstractPlaceProduct):
+    price_unit = EnumField(
```

Comment:
this field is probably handled by `AbstractPlaceProduct`?

Reply:
Right! I forgot to remove it from here. Good catch 👍